app/scripts/BarTrack.js

```diff
@@ -204,24 +204,30 @@ class BarTrack extends HorizontalLine1DPixiTrack {
     super.rerender(options, force);
   }
 
+  getXScaleAndOffset(drawnAtScale) {
+    const dA = drawnAtScale.domain();
+    const dB = this._xScale.domain();
+
+    // scaling between tiles
+    const tileK = (dA[1] - dA[0]) / (dB[1] - dB[0]);
+
+    const newRange = this._xScale.domain().map(drawnAtScale);
+
+    const posOffset = newRange[0];
+
+    return [tileK, -posOffset * tileK];
+  }
+
   draw() {
     // we don't want to call HorizontalLine1DPixiTrack's draw function
     // but rather its parent's
     super.draw();
 
     Object.values(this.fetchedTiles).forEach((tile) => {
-      // scaling between tiles
-      const tileK = (
-        (tile.drawnAtScale.domain()[1] - tile.drawnAtScale.domain()[0])
-        / (this._xScale.domain()[1] - this._xScale.domain()[0])
-      );
-
-      const newRange = this._xScale.domain().map(tile.drawnAtScale);
-
-      const posOffset = newRange[0];
+      const [xScale, xPos] = this.getXScaleAndOffset(tile.drawnAtScale);
```

Comment:
```suggestion
      const [graphicsXScale, graphicsXPos] = this.getXScaleAndOffset(tile.drawnAtScale);
```
Changing these names to more accurately reflect what they represent. `xPos` is ok, but `xScale` is used in many other places and could be confusing if used in another context here.